scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/AzureBlobDownloader.java

```diff
@@ -0,0 +1,222 @@
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.examples;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.channels.AsynchronousFileChannel;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.Map;
+
+import org.ow2.proactive.scheduler.common.task.TaskResult;
+import org.ow2.proactive.scheduler.common.task.executable.JavaExecutable;
+
+import com.microsoft.azure.storage.blob.BlobRange;
+import com.microsoft.azure.storage.blob.BlockBlobURL;
+import com.microsoft.azure.storage.blob.ContainerURL;
+import com.microsoft.azure.storage.blob.ListBlobsOptions;
+import com.microsoft.azure.storage.blob.models.BlobItem;
+import com.microsoft.azure.storage.blob.models.ContainerListBlobFlatSegmentResponse;
+import com.microsoft.rest.v2.util.FlowableUtil;
+
+import io.reactivex.Single;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 13/09/2018
+ */
+public class AzureBlobDownloader extends JavaExecutable {
+
+    private String outputPath;
+
+    private String containerName;
+
+    private String blobName;
+
+    private String accountName;
+
+    private String accountKey;
+
+    private static final String OUTPUT_PATH = "outputPath";
+
+    private static final String CONTAINER_NAME = "containerName";
+
+    private static final String BLOB_NAME = "blobName";
+
+    private ContainerURL containerURL;
+
+    @Override
+    public void init(Map<String, Serializable> args) throws Exception {
+
+        if (args.containsKey(CONTAINER_NAME) && !args.get(CONTAINER_NAME).toString().isEmpty()) {
+            containerName = args.get(CONTAINER_NAME).toString();
+        } else {
+            throw new IllegalArgumentException("You have to specify a container name. Empty value is not allowed.");
+        }
+
+        if (args.containsKey(BLOB_NAME) && !args.get(BLOB_NAME).toString().isEmpty()) {
+            blobName = args.get(BLOB_NAME).toString();
```

Comment:
With blobName as an Optional<String> you could have: 
blobName = Optional.ofNullable(args.get(BLOB_NAME)).filter( blob -> blob.toString().isEmpty());